kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/rocksdb/RocksDBCacheProvider.java

```diff
@@ -54,38 +60,48 @@ public static RocksDBCacheProvider get() {
     return PROVIDER;
   }
 
+  private CacheWrapper getCacheWrapper(String applicationId) {
+    return cacheConfigPerApplicationId.computeIfAbsent(applicationId,
+        s -> new CacheWrapper());
+  }
+
   protected synchronized void initCache(Options options, Map<String, Object> configs) {
-    if (cache == null) {
-      cacheTotalCapacity = DEFAULT_CACHE_TOTAL_CAPACITY;
+    String applicationId = (String) configs.get("application.id");
```

Comment:
Can this `applicationId` be null?

Reply:
`group.id` is mandatory in kafka streams - can't be null.